pyparsing/diagram/__init__.py

```diff
@@ -355,6 +368,15 @@ def _to_diagram_element(
     elif isinstance(element, pyparsing.Empty) and not element.customName:
         # Skip unnamed "Empty" elements
         ret = None
+    elif _is_named_terminal(element):
+        if not _already_extracted(element, lookup):
+            ret = EditablePartial.from_call(
+                railroad.Terminal, title=element.name, text=element.defaultName
+            )
+        else:
+            ret = EditablePartial.from_call(
```

Comment:
I think this check (`if not _already_extracted`) could be simplified just by adding a new condition to `_worth_extracting`. As it stands, we already check if it's `_worth_extracting`, and if it has already been extracted, then I already have code for creating a `NonTerminal` here: https://github.com/pyparsing/pyparsing/blob/1b86c4915003ed6bd9d47126f9337801be3853b0/pyparsing/diagram/__init__.py#L332-L345.

So if you simply add `if _is_named_terminal(element): return True` to `_already_extracted`, I think you get this behaviour for free, with less and simpler code.